src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs

```diff
@@ -253,12 +246,39 @@ public void DrawInnerShadow(ShadowPaintState state, SKCanvas canvas, SKPaint pai
 
 			if (_logger.IsEnabled(LogLevel.Trace))
 			{
-				_logger.Trace($"[ShadowContainer] DrawInnerShadow => strokeWidth: {paint.StrokeWidth}, cornerRadius: {cornerRadius}, x: {offsetX}, y: {offsetY}, width: {shape.Rect.Width}, height: {shape.Rect.Height}");
+				_logger.Trace($"[ShadowContainer] DrawInnerShadow => strokeWidth: {paint.StrokeWidth}, x: {offsetX}, y: {offsetY}, width: {shape.Rect.Width}, height: {shape.Rect.Height}");
 			}
 			canvas.DrawRoundRect(shape, paint);
 		}
 	}
 
+	private record CornerRadiusRectShadowShapeContext(double Width, double Height, CornerRadius CornerRadius) : RoundRectShadowShapeContext(Width, Height)
+	{
+		protected override SKRoundRect GetContentShape(ShadowPaintState state)
+		{
+			var rect = new SKRect(0, 0, (float)Width * state.PixelRatio, (float)Height * state.PixelRatio);
+			var radii = new[] { CornerRadius.TopLeft, CornerRadius.TopRight, CornerRadius.BottomRight, CornerRadius.BottomLeft }
+				.Select(x => (float)x * state.PixelRatio)
+				.Select(x => new SKPoint(x, x))
+				.ToArray();
```

Comment:
Linq statements are generally expensive, it's best to change it to a predefined array. You can make another PR to change, so we avoid further delays on this PR.

Reply:
PR opened https://github.com/unoplatform/uno.toolkit.ui/issues/789